### ISSUE TITLE
A question on block size and multilingual

### ISSUE THREAD

**Issue by Walid-Ahmed** [z-lab/dflash#8]
Thanks a lot for the great work. I have a couple of questions:

1-How did you decide to use block size 16 as the default?
2-For multilingual scenarios, do you plan to use curriculum learning (for example, training on one language first, then the other, or alternating between languages)?
3-The acceptance length is using the target model output as ground truth, is this the case in training as well?

Thanks

**Comment by Walid-Ahmed** [comment]
Thanks a lot for your detailed and informative message.

One more question: did you train with cross-entropy (CE) or KL divergence (KL)? Which one did you end up recommending, and why?

**Comment by jianc99** [comment]
@Walid-Ahmed We tested both CE and KL losses and found that the resulting acceptance lengths were very similar. However, in some narrow scenarios, CE loss converged noticeably faster. In the end, we used CE loss for training the currently released DFlash models, together with a weighted loss decay. We believe there is still room to further optimize the loss formulation, and we plan to explore this in future work.
